tiledb/sm/filter/bitsort_filter.h

```diff
@@ -0,0 +1,278 @@
+/**
+ * @file   bitsort_filter.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines the bit sort filter class. The bitsort filter takes in
+ * any fixed size, non-nullable input data and sorts it according to its
+ * input values, as integers, on write. Then on read, this filter restores
+ * the original order of the data.
+ *
+ * In order to prevent storing positions on write, this filter instead
+ * rewrites the dimension tiles, and uses the dimension tile information
+ * to store the positions in the sort. Then, in the read, the bitsort filter
+ * will sort the data in global order.
+ */
```

Comment:
hmm I'd prefer the first sentence be changed to " * This file declares the bitsort filter class." instead of defines, because I feel like that is less accurate. Thoughts?

Reply:
I think we have the same sentence :)